lib/trento/application/usecases/sap_systems/health_summary_service.ex

```diff
@@ -60,44 +62,19 @@ defmodule Trento.SapSystems.HealthSummaryService do
     |> HealthService.compute_aggregated_health()
   end
 
-  @spec compute_clusters_health(instance_list) :: Health.t()
-  defp compute_clusters_health(instances) do
-    instances
-    |> reject_unclustered_instances()
-    |> clusters_from_instance()
-    |> keep_only_hana_scale_up_clusters()
-    |> health_from_cluster()
-    |> HealthService.compute_aggregated_health()
-  end
-
-  @spec clusters_from_instance(instance_list) :: [ClusterReadModel.t()]
-  defp clusters_from_instance(instances) do
-    instances
-    |> Enum.filter(fn %{host: host} -> host end)
-    |> Enum.map(fn %{host: %{cluster_id: cluster_id}} -> cluster_id end)
-    |> Enum.uniq()
-    |> Enum.map(fn cluster_id -> Repo.get!(ClusterReadModel, cluster_id) end)
-  end
-
-  @spec health_from_cluster([ClusterReadModel.t()]) :: [String.t()]
-  defp health_from_cluster(clusters) do
-    Enum.map(clusters, fn %ClusterReadModel{health: health} -> health end)
-  end
-
-  @spec reject_unclustered_instances(instance_list) :: instance_list
-  defp reject_unclustered_instances(instances) do
-    Enum.reject(instances, fn
-      %{host: %{cluster_id: nil}} -> true
-      _ -> false
-    end)
-  end
-
-  @spec keep_only_hana_scale_up_clusters([ClusterReadModel.t()]) :: [ClusterReadModel.t()]
-  defp keep_only_hana_scale_up_clusters(clusters) do
-    Enum.filter(clusters, fn
-      %ClusterReadModel{type: ClusterType.hana_scale_up()} -> true
-      _ -> false
-    end)
+  @spec compute_cluster_health(instance_list) :: Health.t()
+  defp compute_cluster_health(instances) do
+    cluster_id =
+      Enum.find_value(instances, nil, fn
```

Comment:
If I am getting this right, it means that among a list of either application or database instances, we might or might not find a cluster id meaning the instance is part of a cluster (hana/netweaver) or not.

Reply:
Yes!
This is to cover unclustered/clustered scenarios.
An unclustered app/db has `unknown` as health